src/edu/csus/ecs/pc2/ui/ReportPane.java

```diff
@@ -486,9 +486,9 @@ protected void createHeader(PrintWriter printWriter, IReport report) {
             resumeTime = getContest().getContestTime().getResumeTime();
         }
         if (resumeTime == null) {
-            printWriter.print("  Contest date/time: never started");
+            printWriter.print("  Contest date/time: never started on site "+getContest().getSiteNumber());
```

Comment:
It looks to me like this new line could now throw an uncaught NPE.  Specifically, the previous lines start with resumeTime being set to null, then _if `getContest()` is not null_, resumeTime is updated from the value of `getContest().getContestTime().getResumeTime()`.  So when THIS statement is reached, _`getContest()` could be returning null_.  If that happens, _`getContest().getSiteNumber()` will throw an NPE_.

Reply:
if getContest() return'ed null that would thrown a NPE in the bit before the == null.

Reply:
I don't believe that's true.  The "bit before the == null" is only executed if getContest() != null.  So assume getContest() DOES return null.  Then when it gets to line 488, resumeTime will (still) be null, so it will execute line 489 -- with a null value for getContest().

Reply:
It is possible for a NPE.  It is very very unlikely.

Reply:
Sooo... "very very unlikely" is a criterion for allowing code that COULD cause an NPE?  That seems like a strange position coming from someone who is constantly harping that we should always be doing "continual improvement".  In fact, it sounds more like a "Tammy" comment :)    However, I'm going to let it go based on the fact that there is already a much earlier call to getContest() in this method (see line 476) which is (also) not protected by any handling of an NPE on getContest().    Fixing THAT one would probably require significant additional effort and is indeed "outside the scope" of this Issue.